lib/solvers/revised_primal_solver/simplex.cpp

```diff
@@ -0,0 +1,236 @@
+#include "simplex.hpp"
+#include <algorithm>
+#include <core/consts.hpp>
+#include <iostream>
+#include <iterator>
+
+namespace revised_primal_simplex {
+
+RevisedPrimalSimplex::RevisedPrimalSimplex() {}
+
+void RevisedPrimalSimplex::setProblem(
+    const std::vector<std::vector<float>> table) {
```

Comment:
When I put inputs on the other line I was doing it like:

void RevisedPrimalSimplex::setProblem(
    const std::vector<std::vector<float>> table
) {

Because I think it looks nicer, but completely fine leaving it like this!

Reply:
I'm surprised that would compile since you're not explicitly giving a type to the elements of the vector (how would it know the  memory address of the fourth element if it didn't know how much space each element before it takes?). This feels like it would lead to undefined behavior (UB) so although it's ugly, probably best to stick to `std::vector<std::vector<float>>`. 

If you specifically wanted something that looks better, we could place something along the lines of `using FloatMatrix = std::vector<std::vector<float>>;` in consts.hpp and include it everywhere it's used. This wouldn't slow down the program I don't think since the compiler would essentially convert it back in the binary I believe but I feel like this is more of an aesthetic change that we can sort out later?